apache/incubator-pegasus
# fix: catch exception if unmarshall the rpc request encounters error

### What problem does this PR solve? 
In previous implementation, the cluster will crash if we receive a dirty muation log, and will not recover unless the dirty logs are clear. For more details: https://github.com/apache/incubator-pegasus/issues/617
So in pr(https://github.com/XiaoMi/rdsn/pull/863), I changed the way from coredump to throw exception, and we can catch the exception to avoid cluster crash. 

### Checklist 

##### Tests 

- Manual test (add detailed scripts or steps below)
1. send a dirty request by java client
2. to see whether the perf counter increase
```[replica*app.pegasus*recent_corrupt_write_count@2.4, VOLATILE_NUMBER, 1.00]```
3. to see whether there is a log about this situation.
```pegasus_server_write.cpp:106:on_batched_writes(): pegasus batch writes handler failed, from ={ip}:{port}, exception =no more data to read after end-of-buffer```
